File: client/src/app/modules/miscellaneous/services/special.offer.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { SpecialOffer } from '../../miscellaneous/model/SpecialOffer';

@Injectable({
  providedIn: 'root'
})
export class SpecialOfferService extends AppRestService<SpecialOffer>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getSpecialOffers() {
    return this.getListCached(new HttpParams(), "special-offers");
  }

  addOrUpdateSpecialOffer(specialOffer: SpecialOffer) {
    this.clearListCache(new HttpParams(), "special-offers");
    return this.addOrUpdate(new HttpParams(), "special-offer", specialOffer, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/SpecialOfferService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.SpecialOffer;

public interface SpecialOfferService {
    public List<SpecialOffer> getSpecialOffers();

    public SpecialOffer getSpecialOffer(Integer id);

    public SpecialOffer addOrUpdateSpecialOffer(SpecialOffer specialOffer);
}
